processor/manager/src/main/java/com/redhat/service/bridge/processor/actions/ActionConfiguratorImpl.java

```diff
@@ -0,0 +1,56 @@
+package com.redhat.service.bridge.processor.actions;
+
+import java.util.Collection;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Instance;
+import javax.inject.Inject;
+
+import com.redhat.service.bridge.infra.exceptions.definitions.user.ActionProviderException;
+
+@ApplicationScoped
+public class ActionConfiguratorImpl implements ActionConfigurator {
+
+    @Inject
+    Instance<ActionValidator> validators;
+    @Inject
+    Instance<ActionResolver> resolvers;
+    @Inject
+    Instance<ActionConnector> connectors;
+
+    @Override
+    public ActionValidator getValidator(String actionType) {
+        return getOptional(validators, actionType)
+                .orElseThrow(() -> new ActionProviderException(String.format("No validator found for action type '%s'", actionType)));
+    }
+
+    @Override
+    public Optional<ActionResolver> getResolver(String actionType) {
+        return getOptional(resolvers, actionType);
+    }
+
+    @Override
+    public Optional<ActionConnector> getConnector(String actionType) {
+        return getOptional(connectors, actionType);
+    }
+
+    private static <T extends ActionBean> Optional<T> getOptional(Instance<T> instances, String actionType) {
+        return instances.stream()
+                .filter(a -> a.accept(actionType))
+                .findFirst();
+    }
+
+    Collection<ActionValidator> getValidators() {
```

Comment:
where these functions are used?

Reply:
They're used in `ActionConfiguratorImplTest` to ensure that there are no unexpected beans that are not covered by that test.

Reply:
we generally don't write code to satisfy test case.